src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,298 @@
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    // The presence of *any* disabled value makes the component disabled, *except* for false.
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(value: any) {
+    if (this._value != value) {
+      // Set this before proceeding to ensure no circular loop occurs with selection..
+      this._value = value;
+
+      // Update selected if different from new value.
+      let isAlreadySelected = this._selected != null && this._selected.value == value;
+      if (this._radios != null && !isAlreadySelected) {
+        let matched = this._radios.filter((radio: MdRadioButton) => {
+          return radio.value == value;
+        });
+
+        if (matched.length == 0) {
+          // Didn't find a button that matches this value, return early without setting.
+          return;
+        }
+
+        // Change the selection immediately.
+        this.selected = matched[0];
+      }
+
+      let event = new MdRadioChange();
+      event.source = this._selected;
+      event.value = value;
+      this.change.emit(event);
+    } else {
+      this._value = value;
+    }
+  }
+
+  @Input()
+  get selected() {
+    return this._selected;
+  }
+
+  set selected(selected: MdRadioButton) {
+    this._selected = selected;
+    this.value = selected.value;
+
+    selected.checked = true;
+  }
+}
+
+
+@Component({
+  selector: 'md-radio-button',
+  templateUrl: './components/radio/radio_button.html',
+  styleUrls: ['./components/radio/radio-button.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdRadioButton {
+  @HostBinding('class.md-radio-focused')
+  private _isFocused: boolean;
+
+  /** Whether this radio is checked. */
+  private _checked: boolean = false;
+
+  /** The unique ID for the radio button. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** Analog to HTML 'name' attribute used to group radios for unique selection. */
+  @Input()
+  name: string;
+
+  /** Whether this radio is disabled. */
+  private _disabled: boolean;
+
+  /** Value assigned to this radio.*/
+  private _value: any = null;
+
+  /** The parent radio group. May or may not be present. */
+  radioGroup: MdRadioGroup;
+
+  /** Dispatcher for coordinating radio unique-selection by name. */
+  radioDispatcher: MdRadioDispatcher;
+
+  /** Event emitted when the group value changes. */
+  @Output()
+  change: EventEmitter<MdRadioChange> = new EventEmitter();
+
+  constructor(@Optional() radioGroup: MdRadioGroup, @Attribute('id') id: string,
+              radioDispatcher: MdRadioDispatcher) {
+    // Assertions. Ideally these should be stripped out by the compiler.
+    // TODO(jelbourn): Assert that there's no name binding AND a parent radio group.
+
+    this.radioGroup = radioGroup;
+    this.radioDispatcher = radioDispatcher;
+
+    this.id = id != null ? id : `md-radio-${_uniqueIdCounter++}`;
```

Comment:
You could set `this.id` in `ngOnInit` and then not have to inject `@Attribute('id')` in tbe constructor.